lib/improver/tests/database/test_VerificationDatabase.py

```diff
@@ -0,0 +1,214 @@
+                   "exp_id", "fcr_tplus000", "fcr_tplus001", "fcr_tplus002"]
+        expected_df = pd.DataFrame(data, columns=columns)
+        expected_df = expected_df.set_index(["validity_date", "validity_time",
+                                             "station_id", "cf_name",
+                                             "exp_id"])
+        expected_df.columns.name = "forecast_period"
+        # Call the plugin.
+        cubes = [set_up_spot_cube(
+                    280+i, validity_time=1487311200+3600*i,
+                    forecast_period=i, number_of_sites=1) for i in range(3)]
+        cubes = iris.cube.CubeList(cubes)
+        plugin = VerificationTable("output", "csv", "improver",   "time", "IMPRO", 0)
+        plugin.to_dataframe(cubes, "index")
+        result = plugin.df
+        assert_frame_equal(expected_df, result)
+
+    def test_single_cube_max_lead_time(self):
+        """Basic test using one input cube with larger max lead time in output"""
+        # Set up expected dataframe.
+        validity_date = dt.utcfromtimestamp(1487311200).date()
+        data = [[validity_date, 600, 1000, "air_temperature", "IMPRO", 280., np.nan]]
+        columns = ["validity_date", "validity_time", "station_id", "cf_name",
+                   "exp_id", "fcr_tplus000", "fcr_tplus001"]
+        expected_df = pd.DataFrame(data, columns=columns)
+        expected_df = expected_df.set_index(["validity_date", "validity_time",
+                                             "station_id", "cf_name",
+                                             "exp_id"])
+        expected_df.columns.name = "forecast_period"
+        # Call the plugin.
+        cubes = iris.cube.CubeList([set_up_spot_cube(280, number_of_sites=1)])
+        plugin = VerificationTable("output", "csv", "improver",  "time", "IMPRO", 3600)
+        plugin.to_dataframe(cubes, "index")
+        result = plugin.df
+        assert_frame_equal(expected_df, result)
+
+
+class Test_ensure_all_pivot_columns(IrisTest):
+    """A set of tests for the ensure_all_pivot_collumns method"""
+    def test_single_cube(self):
+        """Basic test using one input cube."""
+
+        cubes = iris.cube.CubeList([set_up_spot_cube(280)])
+        plugin = VerificationTable("csv", "output", "improver", "time",
+                                   "nbhood", 3600)
+        test_dataframe = pd.DataFrame(data=np.array([280.0, 280.0, 280.0]),
+                                      columns=["fcr_tplus000"])
+        plugin.ensure_all_pivot_columns(test_dataframe)
+        result = test_dataframe
+        expected_dataframe = pd.DataFrame(data=np.array([[280.0, np.nan],
+                                                         [280.0, np.nan],
+                                                         [280.0, np.nan]]),
+                                          columns=["fcr_tplus000",
+                                                   "fcr_tplus001"])
+        assert_frame_equal(expected_dataframe, result)
+
```

Comment:
I think this tests the same method as the previous, namely:
 Test_to_dataframe.test_single_cube_max_lead_time()
I think since you've abstracted this from the to_dataframe() method, the previous test may not be required.